src/Device.Net/ControlTransferResult.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace Device.Net
+{
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+    public readonly struct ControlTransferResult
+#pragma warning restore CA1815 // Override equals and operator equals on value types
+    {
+        #region Public Properties
+#pragma warning disable CA1819 // Properties should not return arrays
+        public byte[] Data { get; }
+#pragma warning restore CA1819 // Properties should not return arrays
+
+        public uint BytesTransferred { get; }
+        #endregion
+
+        #region Conversion Operators
+        public byte[] ToByteArray() => Data;
```

Comment:
ReadResult has an implicit operator for converting to arrays

Reply:
Yes, but the code analyser complains about that....
I'm quite all right if you want to default to that implicit operation here too.

Reply:
We can potentially rename the class, but I'm not going to have two types that do exactly the same thing

Reply:
@josesimoes what's the issue with analyzer?

Reply:
It is still ReadResult whether it's from control transfer or a normal read.

Reply:
@josesimoes in my branch I fixed the code analyzer rule.

csharp_style_expression_bodied_operators = true:error

Reply:
That's good!